### PR TITLE
gh-107427: Update the description of UNPACK_SEQUENCE

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-107427

----
:books: Documentation preview :books:: https://cpython-previews--107429.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/dis.rst [hunk 1]
```diff
@@ -924,7 +924,7 @@ iterations of the loop.
    Unpacks ``STACK[-1]`` into *count* individual values, which are put onto the stack
    right-to-left::
 
-      STACK.extend(STACK.pop()[:count:-1])
+      STACK.extend(STACK.pop()[:-count-1:-1])
```
Comment (gvanrossum) [inline Doc/library/dis.rst]: Shouldn't we also document somehow that we require `STACK.pop()` to have exactly `count` elements?

Comment (corona10) [reply Doc/library/dis.rst]: Update the sentence ```Unpacks ``STACK[-1]`` into *count* individual values``` into ```Unpacks ``STACK.pop()`` into *count* individual values``` will be enough?

Comment (gvanrossum) [reply Doc/library/dis.rst]: I don’t think so. Leave that sentence alone, but add e.g. “Require there to be exactly `count` values.”

Comment (corona10) [reply Doc/library/dis.rst]: Done!

**Review by gvanrossum — APPROVED** [review]

**Review by gvanrossum — APPROVED** [review]

**Comment by gvanrossum** [comment]
Technically the thing on top of the stack may not have a `len()`; it can be any iterable. The length check is done (unless it's a tuple or list) by continuing to iterate and expecting `StopIteration`.

**Pull request merged**